### ISSUE TITLE
Are there plans to include gamma blending and stem darkening for rendering?

### ISSUE THREAD

**Issue by kkanungo17** [pop-os/cosmic-text#195]
As described by this freetype article: https://freetype.org/freetype2/docs/hinting/text-rendering-general.html

**Comment by xalt7x** [comment]
Related to: 
-  [iced-rs "web-colors" feature flag PR](https://github.com/iced-rs/iced/pull/1888)
- "[COSMIC: More Alpha, More Fun!](https://blog.system76.com/post/cosmic-more-alpha-more-fun)" (screenshot shows the usage of "web-colors" in cosmic-text).

**Comment by kkanungo17** [comment]
Yes, we finally got a DE with gamma blending for text. Now just waiting on stem darkening.

**Comment by xalt7x** [comment]
> Now just waiting on stem darkening.

I'd like it as well but unfortunately as of now stem-darkening rendering might not look good for some OTF fonts
For example, see issues with Noto Sans OTF:
- https://bugzilla.opensuse.org/attachment.cgi?id=859236
- https://bugreports.qt.io/browse/QTBUG-95336

Pop!_OS currently uses OTF versions of FiraSans and FiraMono fonts.

I hope that either the developers of Freetype, the maintainers of iced, or the maintainers of Pop-OS will be able to tweak/fix it somehow.

**Comment by kkanungo17** [comment]
I think the problem with those is that the fonts either aren't gamma blended (GTK), or use a value that doesn't match the display (I think Qt uses 1.8 instead of 2.2). I'm just guessing though, let's see what COSMIC devs come up with.

**Comment by kkanungo17** [comment]
@xalt7x Actually taking a closer look, COSMIC actually went from correct rendering to wrong rendering lol. [This feature](https://github.com/iced-rs/iced/pull/1888) actually enables the option to not do blending in linear space and instead do it the "wrong" way to follow existing methods.

I can see why they decided to do it, linear space blending on its own makes fonts too light, but the correct way to fix that is stem darkening, not going back to the old wrong way of doing things.

**Comment by kkanungo17** [comment]
Patrick Walton had reverse engineered the metrics Apple uses for stem darkening, maybe this can be used as a basis to darken text that become light due to gamma correct rendering: https://x.com/pcwalton/status/918991457532354560
